tools/pipelines/templates/build-npm-package.yml

```diff
@@ -1,598 +1,540 @@
+                  taskLintName: '${{ parameters.taskLintName }}'
+                  buildDirectory: '${{ parameters.buildDirectory }}'
+              - '${{ if ne(convertToJson(parameters.taskTest), ''[]'') }}':
+                  - script: |
+                      echo "##vso[task.setvariable variable=startTest]true"
+                    displayName: Start Test
+                  - '${{ each taskTestStep in parameters.taskTest }}':
+                      - template: /tools/pipelines/templates/include-test-task.yml@self
+                        parameters:
+                          taskTestStep: '${{ taskTestStep }}'
+                          buildDirectory: '${{ parameters.buildDirectory }}'
+                  - '${{ if eq(variables[''testCoverage''], true) }}':
+                      - task: Bash@3
+                        displayName: Check for nyc/report directory
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}'
+                          script: >
+                            test -d nyc/report && echo '##vso[task.setvariable
+                            variable=ReportDirExists;]true' || echo 'No
+                            nyc/report directory'
+                        condition: >-
+                          and(succeededOrFailed(), eq(variables['startTest'],
+                          'true'))
+                      - task: Bash@3
+                        displayName: Patch Coverage Results
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}/nyc/report'
+                          script: >
+                            sed -e 's/\(filename=\".*[\\/]external
+                            .*\)"\(.*\)""/\1\&quot;\2\&quot;"/'
+                            cobertura-coverage.xml >
+                            cobertura-coverage-patched.xml
+                        condition: >-
+                          and(succeededOrFailed(),
+                          eq(variables['ReportDirExists'], 'true'))
+                      - task: PublishCodeCoverageResults@1
+                        displayName: Publish Code Coverage
+                        inputs:
+                          codeCoverageTool: Cobertura
+                          summaryFileLocation: >-
+                            ${{ parameters.buildDirectory
+                            }}/nyc/report/cobertura-coverage-patched.xml
+                          reportDirectory: '${{ parameters.buildDirectory }}/nyc/report'
+                          failIfCoverageEmpty: true
+                        condition: >-
+                          and(succeededOrFailed(),
+                          eq(variables['ReportDirExists'], 'true'))
+                  - '${{ if contains(convertToJson(parameters.taskTest), ''tinylicious'') }}': []
```

Comment:
Hmmmm initially I was very concerned that the actual publishing task (as many others) was completely removed here, then I realized all publishing tasks moved to the `templateContext.outputs` on line 421. I don't get the purpose of the remaining line here though, maybe it can be removed?

A general concern about what the 1ES template is forcing us to do: publishing an artifact is now very removed from the place in the pipeline flow where the artifact was produced; makes it harder to reason about it, especially when publishing it should be conditional. I see we had to bring the original condition _and_ the one from line 246 into `templateContext.outputs` so we still publish the tinylicious log only in the same instances where we do today. Feels inconvenient.

I think we should at least add comments where we are being forced to move a publishing task, saying "At this point we want to publish X, see templateContext.outputs below where the actual task is defined".